tests/e2e/onpremises/nodes.tf

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2017-present SIGHUP s.r.l All rights reserved.
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file.
+resource "hcloud_server" "haproxy" {
+  name        = "haproxy-0-${var.ci_number}"
+  image       = "ubuntu-24.04"
+  server_type = "cpx31"
+  location    = "nbg1"
+
+  ssh_keys = [hcloud_ssh_key.ssh_key.id]
+  network {
+    network_id = hcloud_network.network.id
+    ip         = "10.10.1.2"
+  }
+  firewall_ids = [hcloud_firewall.haproxy.id]
+}
+
+resource "null_resource" "init_haproxy" {
+  triggers = {
+    instance_id   = hcloud_server.haproxy.id
+    private_key   = var.private_key
+  }
+
+  connection {
+    type        = "ssh"
+    host        = hcloud_server.haproxy.ipv4_address
+    user        = "root"
+    private_key = var.private_key
+  }
+
+  provisioner "remote-exec" {
+    inline = [ 
+      "echo \"IdentityFile ~/.ssh/id_ed25519\" > /root/.ssh/config ",
+      "echo \"${var.private_key}\" > /root/.ssh/id_ed25519",
+      "chmod 600 /root/.ssh/id_ed25519"
+     ]
+  }
+
+}
+
+resource "hcloud_server" "master" {
```

Comment:
we could name this control-plane instead of master, so we won't have to rename it later when we align the nomenclature

Reply:
You are right, let me change it